spec/linkedlist.js

```diff
@@ -0,0 +1,150 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import LinkedList from '../src/linkedlist'
+
+chai.use(chaiChange)
+
+const generateTestList = () => {
+  const list = new LinkedList()
+
+  list.insert('foo')
+  list.insert('stuff')
+  list.insert('blah')
+  list.insert('foo2')
+  list.insert('stuff2')
+
+  return list
+}
+
+
+describe('LinkedList', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(LinkedList).to.be.a('function')
+  })
+
+  describe('insertFirst', () => {
+
+    it('Inserts a node with the provided data to the head of the list', () => {
+      const aLinkedList = new LinkedList()
+
+      expect(() => aLinkedList.insertFirst('foo'))
+        .to.alter(() => aLinkedList.size(), { from: 0, to: 1 })
+    })
+
+    it('Inserts a node with the provided data to the head of the list', () => {
+      const aLinkedList = new LinkedList()
+
+      aLinkedList.insertFirst('bar')
+
+      expect(() => aLinkedList[0]
+        .to.equal('bar'))
+    })
+  })
+
+  describe('insert()', () => {
+    it('Inserts a node (with the provided data) to the tail of the list', () => {
+      const aLinkedList = generateTestList()
+
+      expect(() => aLinkedList.insert('junk'))
+        .to.alter(() => aLinkedList.size(), { from: 5, to: 6 })
+    })
+
+    it('Inserts a node (with the provided data) to the tail of the list', () => {
+      const aLinkedList = generateTestList()
+
+      aLinkedList.insert('junk')
+
+      expect(aLinkedList.tail.data)
+        .to.equal('junk')
+    })
+  })
+
+  describe('find()', () => {
+    it('Returns -1 if not found', () => {
+      const aLinkedList = generateTestList()
+
+      expect(aLinkedList.find('doesntexist'))
+      .to.equal(-1)
```

Comment:
if these aren't 80 characters wide I'd make them single lines
if you've got to break them up indent the .to.equal line so you can easily see it's a part of the expect statement